File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/security/ClientKeyMessage.scala





















// Generated on 05/03/2013 10:35:01
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ClientKeyMessage extends MessageDeserializer {
  val messageId = 5607

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ClientKeyMessage.messageId

    val key: String

    def serialize(buf: Buffer) {
      buf.writeString(key)
    }

  }

  protected def create(key: String): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val key = buf.readString()

    create(key)
  }

}

object ClientKeyMessage extends ClientKeyMessage {
  case class MessageImpl(key: String) extends Message
  type Target = MessageImpl

  override def create(key: String) = MessageImpl(key)

  def apply(key: String) = create(key)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ClientKeyMessage.java


// Generated on 05/08/2013 19:37:59
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ClientKeyMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5607;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public String key;
    
    public ClientKeyMessage() { }
    
    public ClientKeyMessage(String key) {
        this.key = key;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeString(key);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        key = buf.readString();
    }
    
}
